ndk-make.sh

```diff
@@ -63,12 +52,73 @@ mkdir -p armeabi-v7a
 mkdir -p arm64-v8a
 mkdir -p x86
 mkdir -p x86_64
-cp deltachat-core-rust/target/armv7-linux-androideabi/release/libdeltachat.a armeabi-v7a
-cp deltachat-core-rust/target/aarch64-linux-android/release/libdeltachat.a arm64-v8a
-cp deltachat-core-rust/target/i686-linux-android/release/libdeltachat.a x86
-cp deltachat-core-rust/target/x86_64-linux-android/release/libdeltachat.a x86_64
+
+cd deltachat-core-rust
+
+# fix build on MacOS Catalina
+unset CPATH
+
+if test -z $1; then
+  # Full build
+
+  export RUSTFLAGS="-C lto=on -C embed-bitcode=yes"
+  RELEASE="release"
+else
+  # Fast build
+
+  RELEASE="debug"
+fi
+
+if test -z $1 || test $1 = armeabi-v7a; then
+    echo "-- cross compiling to armv7-linux-androideabi (arm) --"
+    export CFLAGS=-D__ANDROID_API__=16
+    TARGET_CC=armv7a-linux-androideabi16-clang \
+    cargo +`cat rust-toolchain` build --$RELEASE --target armv7-linux-androideabi -p deltachat_ffi
+    cp target/armv7-linux-androideabi/$RELEASE/libdeltachat.a ../armeabi-v7a
+fi
+
+if test -z $1 || test $1 = arm64-v8a; then
+    echo "-- cross compiling to aarch64-linux-android (arm64) --"
+    export CFLAGS=-D__ANDROID_API__=21
+    TARGET_CC=aarch64-linux-android21-clang \
+    cargo +`cat rust-toolchain` build --$RELEASE --target aarch64-linux-android -p deltachat_ffi
+    cp target/aarch64-linux-android/$RELEASE/libdeltachat.a ../arm64-v8a
+fi
+
+if test -z $1 || test $1 = x86; then
+    echo "-- cross compiling to i686-linux-android (x86) --"
+    export CFLAGS=-D__ANDROID_API__=16
+    TARGET_CC=i686-linux-android16-clang \
+    cargo +`cat rust-toolchain` build --$RELEASE --target i686-linux-android -p deltachat_ffi
+    cp target/i686-linux-android/$RELEASE/libdeltachat.a ../x86
+fi
+
+if test -z $1 || test $1 = x86_64; then
+    echo "-- cross compiling to x86_64-linux-android (x86_64) --"
+    export CFLAGS=-D__ANDROID_API__=21
+    TARGET_CC=x86_64-linux-android21-clang \
+    cargo +`cat rust-toolchain` build --$RELEASE --target x86_64-linux-android -p deltachat_ffi
+    cp target/x86_64-linux-android/$RELEASE/libdeltachat.a ../x86_64
+fi
 
 echo -- ndk-build --
+
+cd ..
+# Set the right arch in Application.mk:
+oldDotMk="$(cat Application.mk)"
+
+if test $1; then
+    sed -i "s/APP_ABI.*/APP_ABI := $1/g" Application.mk
```

Comment:
this does not work on mac, results in error "invalid command code A", sed seems to be tricky cross-platform.

maybe we should just generate the whole Application.mk via an `echo` command here? there are only three lines in there.

Reply:
`-i` is really not portable: https://unix.stackexchange.com/questions/13711/differences-between-sed-on-mac-osx-and-other-standard-sed

Can use `ed` or just generate with a here-document.

Reply:
whatever :) as said, also `echo "content" >  Application.mk` would be fine (assuming we can create the files on f-droid)

Reply:
Updated to use sed with temporary file, would be nice if you test on MacOs.